### PR TITLE
feat(component/datatable): add the new DataTable component

### PR DESCRIPTION
**Pull request by tiloyi**
## I have read [the contributing guidelines](https://mozaic.adeo.cloud/Contributing/)

- [x] Yes
- [ ] No

## Does this PR introduce a [breaking change](https://mozaic.adeo.cloud/Contributing/Developers/GitConventions/#breaking-changes-)?

- [ ] Yes
- [x] No

## Describe the changes

Add the new DataTable component

GitHub issue number or Jira issue URL: https://jira.adeo.com/browse/MZC-11

### CODE REVIEW

**Review by mohamedMok — APPROVED** [review]

**Review by caqueste — CHANGES_REQUESTED** [review]
Great!
Just a change, please. It misses some dividers:
- On the bottom of the header (where there are the sort and column titles)
- On the top of the footer 😉

File: src/docs/Components/DataTable/previews/sticky.preview.scss [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+}
+
+.example {
+  padding: $mu125;
+}
+
+.mc-datatable {
+  $parent: &;
+
+  &--sticky {
```
Comment (mohamedMok) [inline src/docs/Components/DataTable/previews/sticky.preview.scss:30]: If the max-height is necessary, maybe you can add it to the doc?

File: packages/styles/settings-tools/_s.datatable.scss [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  $color-primary-01-600,
+  $color-primary-01-500
+);
+
+@mixin mc-datatable-base() {
+  border-collapse: collapse;
+  border-spacing: 0;
+  caption-side: bottom;
+  width: 100%;
+  // border-spacing: 0; // à mettre uniquement si border-collpase: separate;
```
Comment (mohamedMok) [inline packages/styles/settings-tools/_s.datatable.scss:95]: If not needed you may want to remove this comment

File: packages/styles/settings-tools/_s.datatable.scss [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+  // table-layout: auto; // could be fixed;
```
Comment (mohamedMok) [inline packages/styles/settings-tools/_s.datatable.scss:96]: Same as above

File: packages/styles/settings-tools/_s.datatable.scss [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+}
+
+@mixin mc-datatable-head-label() {
+  @include set-font-weight("semi-bold");
+  @include set-font-scale("02", "m"); // 12px / 16px
+
+  color: $color-font-thead;
+  // height: $height-datatable-cell-size-m;
```
Comment (mohamedMok) [inline packages/styles/settings-tools/_s.datatable.scss:104]: Same as above

File: packages/styles/settings-tools/_s.datatable.scss [hunk 5]
```diff
@@ -0,0 +1,125 @@
+@use "sass:math";
+
+$presetAdeo: false;
+@if $preset == "adeo" {
```
Comment (mohamedMok) [inline packages/styles/settings-tools/_s.datatable.scss:4]: 💯

**Review by mohamedMok — APPROVED** [review]
Few comments, nothing blocking! 
Great work @tiloyi 💯

File: packages/styles/settings-tools/_s.datatable.scss (hunk 3 above)
Comment (tiloyi) [reply packages/styles/settings-tools/_s.datatable.scss:96]: fixed

File: src/docs/Components/DataTable/previews/sticky.preview.scss (hunk 1 above)
Comment (tiloyi) [reply src/docs/Components/DataTable/previews/sticky.preview.scss:30]: You are right. I just fixed it.

File: packages/styles/settings-tools/_s.datatable.scss (hunk 2 above)
Comment (tiloyi) [reply packages/styles/settings-tools/_s.datatable.scss]: fixed

File: packages/styles/settings-tools/_s.datatable.scss (hunk 4 above)
Comment (tiloyi) [reply packages/styles/settings-tools/_s.datatable.scss]: fixed

**Review by jeanBptst — APPROVED** [review]

**Review by caqueste — APPROVED** [review]

**Pull request merged**